packages/manager/apps/pci-ai-notebooks/src/hooks/api/ai/notebook/useStopNotebook.hook.tsx

```diff
@@ -0,0 +1,34 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useParams } from 'react-router-dom';
+import { AIError, NotebookData } from '@/data/api';
+import { stopNotebook } from '@/data/api/ai/notebook/notebook.api';
+
+interface UseStopNotebook {
+  onError: (cause: AIError) => void;
+  onStopSuccess: () => void;
+}
+
+export function useStopNotebook({ onError, onStopSuccess }: UseStopNotebook) {
+  const queryClient = useQueryClient();
+  const { projectId } = useParams();
+  const mutation = useMutation({
+    mutationFn: (notebookInfo: NotebookData) => {
+      return stopNotebook(notebookInfo);
+    },
+    onError,
+    onSuccess: () => {
+      onStopSuccess();
+      // Invalidate service list query to get the latest data
+      queryClient.invalidateQueries({
+        queryKey: [projectId, 'ai/notebook'],
+      });
```

Comment:
Same here, maybe we need to invalidate the getNotebook query

Reply:
Seen together, invalidateQueries will invalidate all queries starting by ai/notebook